pandas/tests/io/parser/test_read_fwf.py

```diff
@@ -141,6 +141,17 @@ def test_fwf_colspecs_None(self):
         expected = DataFrame([[123456, 456], [456789, 789]])
         tm.assert_frame_equal(result, expected)
 
+    def test_fwf_colspecs_infer_nrows(self):
+        # GH 15138
+        # infer_nrows = 1 should have colspec == [(2, 3), (5, 6)]
+        data = """\
+  1  2
+123 98
+"""
+        df = read_fwf(StringIO(data), header=None, infer_nrows=1)
```

Comment:
can you test when infer_nrows > number of rows

Reply:
Sure thing.